### PR TITLE
Xinetd drop and firewalld adoption

### PR DESCRIPTION
**Pull request by jreidinger**
https://trello.com/c/XSlWEaOg/1283-5-research-the-easiest-conversion-from-xinetd-to-sockets

### CODE REVIEW

File: src/modules/TftpServer.rb [hunk 1]
```diff
@@ -23,36 +32,45 @@ def main
...
+      @sysconfig = ::CFA::TftpSysconfig.new
+      # sysconfig values we are interested in. Allow to change it with UI.
+      # TODO when doing bigger changes use sysconfig model everywhere
+      @directory = "/src/tftpboot" # default value
 
       # Detect who is serving tftp:
       # Inetd may be running, it is the default. But it is ok unless it is
       # serving tftp. So we detect who is serving tftp and warn if it is
-      # not xinetd or in.tftpd.
+      # not socket or in.tftpd.
```
Comment (DimStar77) [inline src/modules/TftpServer.rb]: The reference just above about 'inetd being the default' sounds outdayed

File: src/modules/TftpServer.rb [hunk 2]
```diff
@@ -23,36 +32,45 @@ def main
...
-      @directory = ""
-
-      # Other arguments to in.tftpd, ie. not including -s or /dir
-      @other_args = ""
-
+      # sysconfig model
+      @sysconfig = ::CFA::TftpSysconfig.new
+      # sysconfig values we are interested in. Allow to change it with UI.
+      # TODO when doing bigger changes use sysconfig model everywhere
+      @directory = "/src/tftpboot" # default value
```
Comment (DimStar77) [inline src/modules/TftpServer.rb]: Did you mean ```/srv/tftproot```

Comment (jreidinger) [reply src/modules/TftpServer.rb]: no, I use same default as in https://build.suse.de/package/view_file/SUSE:SLE-15:GA/tftp/tftp.sysconfig?expand=1
where is also tftpboot

Comment (DimStar77) [reply src/modules/TftpServer.rb]: I mean src vs srv

File: src/modules/TftpServer.rb [hunk 3]
```diff
@@ -11,10 +11,19 @@
...
+require "y2firewall/firewalld"
+
+
 module Yast
   class TftpServerClass < Module
 
-    SERVICE_NAME = "xinetd"
+    SOCKET_NAME = "tftp"
+    PACKAGE_NAME = "tftp"
+    SERVER_BIN = "in.tftp"
```
Comment (DimStar77) [inline src/modules/TftpServer.rb]: Just curious: this seems nowhere to be used

Comment (jreidinger) [reply src/modules/TftpServer.rb]: it was in killall and then I change it, thanks for notice

File: src/modules/TftpServer.rb [hunk 4]
```diff
@@ -23,36 +31,45 @@ def main
...
       # If nonempty, the user is notified and the module gives up.
       @foreign_servers = ""
     end
 
+    # firewall instance
+    def firewall
+      @firewall ||= Y2Firewall::Firewalld.instance
+    end
+
+    # systemd socket for tfpt
```
Comment (lslezak) [inline src/modules/TftpServer.rb]: NP: typo `tfpt`

File: test/test_helper.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+ENV["Y2DIR"] = File.expand_path("../../src", __FILE__)
+
+require "yast"
+require "yast/rspec"
+
+# force utf-8 encoding for external
+Encoding.default_external = Encoding::UTF_8
```
Comment (lslezak) [inline test/test_helper.rb]: I think you could alternatively use `ENV["LC_ALL"] = "en_US.UTF-8"` which has an advantage that the translated messages do not break the tests when running in non-English locale.

**Pull request merged**